apps-rendering/src/client/callouts.ts

```diff
@@ -0,0 +1,353 @@
+import type { FormField } from '@guardian/apps-rendering-api-models/formField';
+import type { FormOption } from '@guardian/apps-rendering-api-models/formOption';
+import { ArticlePillar, ArticleSpecial } from '@guardian/libs';
+import type { ArticleFormat, ArticleTheme } from '@guardian/libs';
+import { withDefault } from '@guardian/types';
+import type { Callout } from 'bodyElement';
+import { ElementKind } from 'bodyElementKind';
+import { pipe, resultFromNullable } from 'lib';
+import {
+	andThen,
+	arrayParser,
+	booleanParser,
+	documentFragmentParser,
+	fieldParser,
+	map,
+	map2,
+	map7,
+	maybe,
+	numberParser,
+	parse,
+	stringParser,
+	succeed,
+} from 'parser';
+import type { Parser } from 'parser';
+import { createElement as h } from 'react';
+import ReactDOM from 'react-dom';
+import { Result } from 'result';
+import CalloutComponent from '../components/Callout';
+
+type FormData = Record<string, string | string[]>;
+
+type FormElementType =
+	| HTMLInputElement
+	| HTMLTextAreaElement
+	| HTMLSelectElement;
+
+function readFile(file: Blob): Promise<string> {
+	return new Promise((resolve, reject) => {
+		const reader = new FileReader();
+		setTimeout(reject, 30000);
+
+		reader.addEventListener('load', () => {
+			if (reader.result) {
+				const fileAsBase64 = reader.result
+					.toString()
+					.split(';base64,')[1];
+				resolve(fileAsBase64);
+			}
+		});
+
+		reader.addEventListener('error', () => {
+			reject();
+		});
+
+		reader.readAsDataURL(file);
+	});
+}
+function submitCallout(body: FormData, form: Element): void {
+	fetch(
+		'https://callouts.code.dev-guardianapis.com/formstack-campaign/submit',
+		{
+			method: 'POST',
+			body: JSON.stringify(body),
+		},
+	)
+		.then((res) => {
+			if (res.ok) {
+				form.classList.add('js-callout--success');
+			} else {
+				form.classList.add('js-callout--failure');
+			}
+		})
+		.catch(() => {
+			form.classList.add('js-callout--failure');
+		});
+}
+
+const getValueFromElement = async (
+	element: FormElementType,
+	acc: FormData,
```

Comment:
What does `acc` mean here? I can see it's of type FormData, just wondering if there is a better variable name to improve readability?

Reply:
It stands for `accumulator` and is pretty common terminology when used in a reduce as it is here, given that this vanillaJS submission handler is going to be removed anyway I reckon its ok for now! [Here's an overview of JS Reduce ](https://valuelogic.one/blog/the-power-of-javascripts-reduce-function/#:~:text=Reducer%20is%20a%20function%20that,result%20of%20the%20reduce%20function.)

Reply:
Oops - silly me! Thanks for the link